lib/Model/Type.cpp

```diff
@@ -786,28 +782,44 @@ verifyImpl(VerifyHelper &VH, const StructType *T) {
 
 static RecursiveCoroutine<bool>
 verifyImpl(VerifyHelper &VH, const UnionType *T) {
-  if (not T->CustomName.verify(VH) or T->Kind != TypeKind::Union
-      or T->Fields.empty())
-    rc_return false;
+  revng_assert(T->Kind == TypeKind::Union);
+
+  if (not T->CustomName.verify(VH))
+    rc_return VH.fail("Invalid name", *T);
+
+  if (T->Fields.empty())
+    rc_return VH.fail("Union type has zero fields", *T);
 
   llvm::SmallSet<llvm::StringRef, 8> Names;
   for (auto &Group : llvm::enumerate(T->Fields)) {
     auto &Field = Group.value();
     uint64_t ExpectedIndex = Group.index();
 
-    if (Field.Index != ExpectedIndex)
-      rc_return VH.fail();
+    if (Field.Index != ExpectedIndex) {
+      using llvm::Twine;
```

Comment:
This is used a lot, please put it at the top of the file once.

Reply:
Done